### DISCUSSION TITLE
Question: type-safe proxies

### DISCUSSION THREAD

**Discussion by signor-pedro** [doctrine/doctrine2#8610] [Support Questions] [answered]
Hi, I am using psalm heavily and have reached a situation where I can have
```
public function doSth(EntityInterface|Proxy $entity): bool
{
    //....
}
```
All my entities implement the EntityInterface, because I enforce that every entity has a `getId()` method.

Obviously, my psalm now cries
```
Method Doctrine\Common\Proxy\Proxy::getId does not exist
```

Is there some adviced way to make Proxies type-safe? Thank you

**Comment by beberlei** [comment, marked as answer]
Why do you typehint for proxy? You should not need this. 

Correct typehint is EntityInterface|(EntityInterface&Proxy)

**Comment by signor-pedro** [comment]
Thank you very much- I did not know of the intersection types.
